### PR TITLE
Remove the 4 hour Delay to download new Firmware

### PR DESCRIPTION
**Pull request by alberk8**
## Description

1. Remove the 4 Hour delay to download new Firmware.
2. Always check with Cloudsmith API for the latest or fwversion as provided by user.
3. If the firmware already exists on local disk then do not download from Cloudsmith
4. Added Try Catch to CheckVersion so that it works even offline (without internet)

## Motivation and Context
- This allows for loading firmware version as and when it is needed without the time limitation or resorting to deleting the directory "c:\users\username\.nanoframework\" contents

## How Has This Been Tested?

Get latest stable firmware
nanoff --target ESP32_WROOM_32 --update --serialport COM12

Get an older version of stable firmware
nanoff --target ESP32_WROOM_32 --update --serialport COM12 --fwversion 1.6.5.3

Get latest stable firmware
nanoff --target ESP32_WROOM_32 --update --serialport COM12

Get latest preview firmware
nanoff --target ESP32_WROOM_32 --update --serialport COM12 --preview

Get version 815 preview firmware
nanoff --target ESP32_WROOM_32 --update --serialport COM12 --preview --fwversion 1.7.1-preview.815

Get version 840 preview firmware
nanoff --target ESP32_WROOM_32 --update --serialport COM12 --preview --fwversion 1.7.1-preview.840

Get latest preview firmware
nanoff --target ESP32_WROOM_32 --update --serialport COM12 --preview

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: nanoFirmwareFlasher/Program.cs [hunk 1]
```diff
@@ -90,25 +90,33 @@ private static void CheckVersion()
         {
             Version latestVersion;
             Version currentVersion = Version.Parse(_informationalVersionAttribute.InformationalVersion.Split('+')[0]);
-
-            using (var client = new HttpClient())
+            try
```
Comment (josesimoes) [inline nanoFirmwareFlasher/Program.cs:93]: Let's have the PRs atomic and addressing one issue at the time.
Although relevant, this belong to another one.

Please remove these and make a separate PR.
As you are at it, there is no need to output any warning on failure to connect to github. This is not critical and not doing it is of no concern. So less noise for the console. 😉

File: nanoFirmwareFlasher/FirmwarePackage.cs [hunk 2]
```diff
@@ -146,12 +146,23 @@ protected async System.Threading.Tasks.Task<ExitCodes> DownloadAndExtractAsync()
 
             if (fwFiles.Any())
             {
-                // get file creation date (from the 1st one)
-                if ((DateTime.UtcNow - File.GetLastWriteTimeUtc(fwFiles.First().FullName)).TotalHours < 4)
+                //// get file creation date (from the 1st one)
```
Comment (josesimoes) [inline nanoFirmwareFlasher/FirmwarePackage.cs:149]: Please remove the commented code

File: nanoFirmwareFlasher/FirmwarePackage.cs [hunk 3]
```diff
@@ -146,12 +146,23 @@ protected async System.Threading.Tasks.Task<ExitCodes> DownloadAndExtractAsync()
...
+                //if ((DateTime.UtcNow - File.GetLastWriteTimeUtc(fwFiles.First().FullName)).TotalHours < 4)
+                //{
+                //    // fw package has less than 4 hours
+                //    // skip download
+                //    skipDownload = true;
+                //}
+
+                if (!string.IsNullOrEmpty(_fwVersion))
```
Comment (josesimoes) [inline nanoFirmwareFlasher/FirmwarePackage.cs:157]: Why are these checks necessary? Isn't this just a matter of _not_ checking the file date? 🤔

**Review by josesimoes — CHANGES_REQUESTED** [review]

File: nanoFirmwareFlasher/FirmwarePackage.cs (hunk 3 above)
Comment (alberk8) [reply nanoFirmwareFlasher/FirmwarePackage.cs]: I have removed the whole Block and does not affect the running of the program.

File: nanoFirmwareFlasher/Program.cs (hunk 1 above)
Comment (alberk8) [reply nanoFirmwareFlasher/Program.cs]: Removed from code until next pull request.

File: nanoFirmwareFlasher/FirmwarePackage.cs (hunk 2 above)
Comment (alberk8) [reply nanoFirmwareFlasher/FirmwarePackage.cs]: Removed

**Review by josesimoes — APPROVED** [review]
LGTM! Thanks for taking care of this. 👍🏻

**Pull request merged**